src/Microsoft.DotNet.Cli.Utils/Command.cs

```diff
@@ -53,20 +59,29 @@ public static Command Create(string executable, IEnumerable<string> args, NuGetF
 
         public static Command Create(string executable, string args, NuGetFramework framework = null)
         {
-            ResolveExecutablePath(ref executable, ref args, framework);
+            bool isUnsafe = false;
+            ResolveExecutablePath(ref executable, ref args, ref isUnsafe, framework);
 
-            return new Command(executable, args);
+            return new Command(executable, args, isUnsafe);
         }
 
-        private static void ResolveExecutablePath(ref string executable, ref string args, NuGetFramework framework = null)
+        private static void ResolveExecutablePath(ref string executable, ref string args, ref bool isUnsafe, NuGetFramework framework = null)
         {
-            executable = 
-                ResolveExecutablePathFromProject(executable, framework) ??
-                ResolveExecutableFromPath(executable, ref args);
+            var exeFromProject = ResolveExecutablePathFromProject(executable, framework);
+            if (exeFromProject != null)
+            {
+                isUnsafe = false;
```

Comment:
With changes mentioned above, I would prefer to see `ResolveExecutablePathFromProject` and 'ResolveExecutableFromPath' setting the load type. This function only exists to switch between those two modes.